libnetwork/internal/resolvconf/resolvconf_unix_test.go

```diff
@@ -0,0 +1,817 @@
+//go:build !windows
+
+package resolvconf
+
+import (
+	"bytes"
+	"io/fs"
+	"net/netip"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"gotest.tools/v3/assert"
+	is "gotest.tools/v3/assert/cmp"
+)
+
+func TestRCOption(t *testing.T) {
+	testcases := []struct {
+		name     string
+		options  string
+		search   string
+		expFound bool
+		expValue string
+	}{
+		{
+			name:    "Empty options",
+			options: "",
+			search:  "ndots",
+		},
+		{
+			name:    "Not found",
+			options: "ndots:0 edns0",
+			search:  "trust-ad",
+		},
+		{
+			name:     "Found with value",
+			options:  "ndots:0 edns0",
+			search:   "ndots",
+			expFound: true,
+			expValue: "0",
+		},
+		{
+			name:     "Found without value",
+			options:  "ndots:0 edns0",
+			search:   "edns0",
+			expFound: true,
+			expValue: "",
+		},
+		{
+			name:     "Found last value",
+			options:  "ndots:0 edns0 ndots:1",
+			search:   "ndots",
+			expFound: true,
+			expValue: "1",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			rc, err := Parse("", bytes.NewBuffer([]byte("options "+tc.options)))
+			assert.NilError(t, err)
+			value, found := rc.Option(tc.search)
+			assert.Check(t, is.Equal(found, tc.expFound))
+			assert.Check(t, is.Equal(value, tc.expValue))
+		})
+	}
+}
+
+func TestRCWrite(t *testing.T) {
+	testcases := []struct {
+		name            string
+		fileName        string
+		perm            os.FileMode
+		hashFileName    string
+		modify          bool
+		expUserModified bool
+	}{
+		{
+			name:         "Write with hash",
+			fileName:     "testfile",
+			hashFileName: "testfile.hash",
+		},
+		{
+			name:            "Write with hash and modify",
+			fileName:        "testfile",
+			hashFileName:    "testfile.hash",
+			modify:          true,
+			expUserModified: true,
+		},
+		{
+			name:            "Write without hash and modify",
+			fileName:        "testfile",
+			modify:          true,
+			expUserModified: false,
+		},
+		{
+			name:     "Write perm",
+			fileName: "testfile",
+			perm:     0640,
+		},
+	}
+
+	rc, err := Parse("", bytes.NewBuffer([]byte("nameserver 1.2.3.4")))
+	assert.NilError(t, err)
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			d := t.TempDir()
+			path := filepath.Join(d, tc.fileName)
+			var hashPath string
+			if tc.hashFileName != "" {
+				hashPath = filepath.Join(d, tc.hashFileName)
+			}
+			if tc.perm == 0 {
+				tc.perm = 0644
+			}
+			err := rc.WriteFile(path, hashPath, tc.perm)
+			assert.NilError(t, err)
+
+			fi, err := os.Stat(path)
+			assert.NilError(t, err)
+			assert.Check(t, is.Equal(fi.Mode(), tc.perm))
+
+			if tc.modify {
+				err := os.WriteFile(path, []byte("modified"), 0644)
+				assert.NilError(t, err)
+			}
+
+			um, err := UserModified(path, hashPath)
+			assert.NilError(t, err)
+			assert.Check(t, is.Equal(um, tc.expUserModified))
+		})
+	}
+}
+
+func a2s(addrs []netip.Addr) []string {
```

Comment:
It's unfortunate the standard library does not have a generic `slices.Map` function. (Yet. See https://go.dev/issue/61898) So I whipped up #47209 for you.
```go
var a2s = sliceutil.Mapper(netip.Addr.String)
```

Reply:
Nice! Done.